arch/arc/core/userspace.S

```diff
@@ -98,8 +98,10 @@ SECTION_FUNC(TEXT, z_arc_userspace_enter)
 
 	/* the end of user stack in r5 */
 	add r5, r4, r5
-	/* start of privilege stack */
-	add blink, r5, CONFIG_PRIVILEGED_STACK_SIZE + STACK_GUARD_SIZE
+	/* get start of privilege stack */
+	ld blink, [r6, _thread_offset_to_priv_stack_start]
```

Comment:
Some explanation/assumptions on the parameters passed through registers for this function would be good (r5 is mentioned above, but r6 not).

Reply:
Ok